pkg/clusters/addons/kong/kong.go

```diff
@@ -146,9 +161,34 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		)
 	}
 
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
+	if a.enterprise {
+		imageRepo := fmt.Sprintf("image.repository=%s", a.repo)
+		imageTag := fmt.Sprintf("image.tag=%s", a.tag)
+		a.deployArgs = append(a.deployArgs,
+			"--set", imageRepo,
+			"--set", imageTag,
+		)
+
+		if a.dbmode == PostgreSQL {
```

Comment:
I'm not certain why we're triggering license deployment _only_ for postgres backed deployments, the database mode should not be involved in that decision.

If we require a license when enabling enterprise mode as was requested above:

```
func (b *Builder) WithEnterprise(license string) *Builder {
```

Then we can do this license deployment logic based on `a.enterprise` only which would appear to be ideal.

Reply:
Yes 👍 , agree License should be added as parameter, and generically apply to enterprise deployment.